utils/monitoring_tools/hardware_monitoring/README.md

```diff
@@ -0,0 +1,158 @@
+     ```bash
+     export NEPTUNE_API_TOKEN=your_token
+     export NEPTUNE_PROJECT=your_workspace/your_project
+     ```
+
+3. Download and place `neptune_hardware_monitoring.py` in your project directory.
+
+4. **Integrate the monitor in your script:**  
+   - Using a context manager:
+    ```python
+    from neptune_scale import Run
+    from neptune_hardware_monitoring import SystemMetricsMonitor
+
+    run = Run()
+    with SystemMetricsMonitor(run=run):
+        # Your training or workload code here
+        ...
+    ```
+    - Using `start()` and `stop()` methods:
+    ```python
+    from neptune_scale import Run
+    from neptune_hardware_monitoring import SystemMetricsMonitor
+
+    run = Run()
+    monitor = SystemMetricsMonitor(run=run)
+
+    monitor.start()
+    # Your training or workload code here
+    monitor.stop()
+    ```
+
+5. **Metrics will be logged to Neptune automatically** under the specified namespace.
+
+## API Reference
+
+```python
+class SystemMetricsMonitor(
+    run: Run,
+    sampling_rate: float = 5.0,
+    namespace: str = "runtime",
+)
+```
+- **run**: Neptune Run object for logging metrics.
+- **sampling_rate**: How often to sample metrics (in seconds). Default is `5.0`.
+- **namespace**: [Namespace](https://docs.neptune.ai/namespaces_and_attributes/) where the metrics will be logged in the Neptune run. Default is `"runtime"`.
+
+---
+
+## Caveats
+
+- **GPU monitoring requires `pynvml` and a compatible NVIDIA GPU.** If not available, GPU metrics are skipped and a warning is shown.
+- **File descriptor metrics** are only available on Unix-like systems.
+- **Each process logs its own metrics.** In multi-process scenarios when logging to the same Neptune run, ensure that each process uses a different `namespace`.
+
+---
+
+## Example Output
+
+Details and metrics are logged to Neptune in a structured namespace, e.g.:
+```
+runtime/details/gpu/name/0
+runtime/details/gpu_num
+runtime/monitoring/cpu/percent
+runtime/monitoring/memory/virtual_used_GB
+runtime/monitoring/gpu/0/memory_used_GB
+runtime/monitoring/process/rss_memory_MB
+...
+```
+
+[![Explore in Neptune](https://neptune.ai/wp-content/uploads/2024/01/neptune-badge.svg)](https://scale.neptune.ai/o/examples/org/showcase/runs/details?viewId=9f113328-75aa-4c61-9aa8-5bbdffa90879&detailsTab=attributes&runIdentificationKey=hardware_monitoring&type=experiment&compare=uiR6bnResm5Dc79uAlyMySpGVJXsCp6medoKxSlVtJGQ&path=runtime%2F)
+---
+
+Create [custom dashboards](https://docs.neptune.ai/custom_dashboard) to visualize all important hardware metrics.
+
+__Example dashboard__: [![Explore in Neptune](https://neptune.ai/wp-content/uploads/2024/01/neptune-badge.svg)](https://scale.neptune.ai/o/examples/org/showcase/runs/details?viewId=9f113328-75aa-4c61-9aa8-5bbdffa90879&detailsTab=dashboard&dashboardId=9f11330c-e4ff-413a-9faa-9e10e5b3f7ee&runIdentificationKey=hardware_monitoring&type=experiment&compare=uiR6bnResm5Dc79uAlyMySpGVJXsCp6medoKxSlVtJGQ)
```

Comment:
The badge looks weird here.

Reply:
Same as the top one. Do I remove the badge altogether and use just hyperlinked texts?

Reply:
Sounds good